### PR TITLE
OSSMDOC-312: Installation reorg

### PR DESCRIPTION
**Pull request by neal-timpe**
- Aligned team: Service mesh
- For branches: 4.6, 4.7, 4.8
- https://issues.redhat.com/browse/OSSMDOC-312
- https://issues.redhat.com/browse/OSSMDOC-301
- https://issues.redhat.com/browse/OSSMDOC-304

Links to doc preview and explanation of work:
- Moves information about the operators from [preparing to install](https://deploy-preview-31766--osdocs.netlify.app/openshift-enterprise/latest/service_mesh/v2x/preparing-ossm-installation.html) to [install](https://deploy-preview-31766--osdocs.netlify.app/openshift-enterprise/latest/service_mesh/v2x/installing-ossm.html#ossm-installation-activities_installing-ossm).
- Condenses operator [installation steps](https://deploy-preview-31766--osdocs.netlify.app/openshift-enterprise/latest/service_mesh/v2x/installing-ossm.html#ossm-install-ossm-operator_installing-ossm).
- Moves [SMCP to its own assembly](https://deploy-preview-31766--osdocs.netlify.app/openshift-enterprise/latest/service_mesh/v2x/ossm-create-smcp.html). Jamie's feedback was that SMCP and member roll were similar tasks, but the Operator and SMCP weren't.
- Moves or removes information about unsupported tasks.
- Moves tracing tutorial to [data visualization](https://deploy-preview-31766--osdocs.netlify.app/openshift-enterprise/latest/service_mesh/v2x/ossm-observability.html#generating-sample-traces-analyzing-trace-data_observability).
- Adds assembly for [adding services to a mesh with a member roll](https://deploy-preview-31766--osdocs.netlify.app/openshift-enterprise/latest/service_mesh/v2x/ossm-create-mesh.html). I believe it is better understood as a task separate from the installation steps.
- Splits the old deploying assembly into [enabling sidecar injection](https://deploy-preview-31766--osdocs.netlify.app/openshift-enterprise/latest/service_mesh/v2x/prepare-to-deploy-applications-ossm.html) and [managing profiles and users](https://deploy-preview-31766--osdocs.netlify.app/openshift-enterprise/latest/service_mesh/v2x/ossm-profiles-users.html). [Members](https://deploy-preview-31766--osdocs.netlify.app/openshift-enterprise/latest/service_mesh/v2x/ossm-profiles-users.html#ossm-members_ossm-profiles-users) used to be in the installation steps.
- I have decided to include my review of [deployment](https://deploy-preview-31766--osdocs.netlify.app/openshift-enterprise/latest/service_mesh/v2x/prepare-to-deploy-applications-ossm.html) in this PR. Deploying apps is an OCP/Kube function. I think we need to make that clear in our docs and link to them. I think there is still some work to be done on this assembly, though. 
- Moved Jaeger tutorial to [Data viz](https://deploy-preview-31766--osdocs.netlify.app/openshift-enterprise/latest/service_mesh/v2x/ossm-observability.html#generating-sample-traces-analyzing-trace-data_observability)

- SME review: brian-avery and dgn
- QE review: yxun, gbaufake, and pbajjuri20
- Peer review: hjoy

All reviews completed and approved. Ready to merge.

### CODE REVIEW

File: modules/ossm-automatic-sidecar-injection.adoc [hunk 1]
```diff
@@ -6,20 +6,24 @@
 [id="ossm-automatic-sidecar-injection_{context}"]
 = Enabling automatic sidecar injection
 
-When deploying an application into the {ProductName} you must opt in to injection by specifying the `sidecar.istio.io/inject` annotation with a value of `"true"`. Opting in ensures that the sidecar injection does not interfere with other OpenShift features such as builder pods used by numerous frameworks within the OpenShift ecosystem.
+When deploying an application, you must opt-in to injection by setting the `sidecar.istio.io/inject` annotation to `"true"`. Opting in ensures that the sidecar injection does not interfere with other OpenShift features such as builder pods used by numerous frameworks within the OpenShift ecosystem. Automatic injection using annotation removes the need to label projects.
```
Comment (brian-avery) [inline modules/ossm-automatic-sidecar-injection.adoc:9]: Is "removes the need to label projects" appropriate here? 

From an OCP point of view -- labeling projects requires additional privileges that not all users have so in that sense "removes the need" is correct

From the point of view of someone who knows upstream Istio -- the user is likely familiar with labeling projects to have all pods in that project injected with the sidecar. In this case, we aren't removing the need for anything, but rather using a different behavior with the labeling of the project not being supported.

File: modules/ossm-automatic-sidecar-injection.adoc [hunk 2]
```diff
@@ -6,20 +6,24 @@
...
 
 .Prerequisites
 
 * Identify the deployments for which you want to enable automatic sidecar injection.
-* Locate the application's YAML configuration file.
 
 .Procedure
 
-. Open the application's configuration YAML file in an editor.
+. Open the application's deployment configuration YAML file in an editor. To find a deployment use the `oc get` command. For example, for an app called `sleep` in the `sleep` namespace, use the following command to see the resource in YAML format.
```
Comment (brian-avery) [inline modules/ossm-automatic-sidecar-injection.adoc:17]: Should this be (as specified below):

Open the application's deployment configuration YAML in an editor.

It's not really a file in Kubernetes, but rather a resource. Also, would it simplify things to do:

```
oc edit deployment sleep -o yaml 
```

This would open the resource in the system's text editor without having to save to a file and apply after. 

Regardless,

```
oc get deployment sleep - yaml
```

is missing the -o. It should be:

```
oc get deployment sleep - yaml
```

File: modules/ossm-control-plane-cli.adoc [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+* An account with the `cluster-admin` role.
+* Access to the OpenShift CLI (`oc`).
+
+.Procedure
+
+. Log in to the {product-title} CLI as a user with the `cluster-admin` role.
++
+[source,terminal]
+----
+$ oc login https://{HOSTNAME}:6443
```
Comment (brian-avery) [inline modules/ossm-control-plane-cli.adoc:22]: This is outside of the scope of OSSM and really part of setting up oc so you can use your cluster. Should it be included here or assumed?

File: modules/ossm-control-plane-web.adoc [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+. Create a project named `istio-system`.
++
+.. Navigate to *Home* -> *Projects*.
++
+.. Click *Create Project*.
++
+.. In the *Name* field, enter `istio-system`. The `ServiceMeshControlPlane` resource must be installed in a project that is separate from your microservices and operators.
++
+These steps use `istio-system` project as an example, but you can deploy your control plane in any namespace as long as it is separate from the project that contains your services.
```
Comment (brian-avery) [inline modules/ossm-control-plane-web.adoc:27]: These steps use the `istio-system` project as an example

File: service_mesh/v2x/ossm-create-mesh.adoc [hunk 5]
```diff
@@ -0,0 +1,31 @@
+[id="ossm-create-mesh"]
+= Adding services to a service mesh
+include::modules/ossm-document-attributes.adoc[]
+:context: ossm-create-mesh
+
+After installing the operators and `ServiceMeshControlPlane` resource, add applications, workloads, or services to your mesh by creating a `ServiceMeshMemberRoll` resource and specifying the namespaces where your content is located. If you already have an application, workflow, or service to add to a `ServiceMeshMemberRoll` resource, use the following steps. Or, to install a sample application called Bookinfo and add it to a  `ServiceMeshMemberRoll` resource, skip to the tutorial for installing the xref:../../service_mesh/v2x/ossm-create-mesh.adoc#ossm-tutorial-bookinfo-overview_ossm-create-mesh[Bookinfo example application] to see how an application works in {ProductName}.
+
+The items listed in the `ServiceMeshMemberRoll` resource are the applications and workflows that are managed by the `ServiceMeshControlPlane` resource. The control plane, which includes the {ProductShortName} Operators, Istiod, and `ServiceMeshControlPlane`, and the data plane, which includes applications and Envoy proxy must be in separate namespaces.
```
Comment (brian-avery) [inline service_mesh/v2x/ossm-create-mesh.adoc:8]: Should this have a comma? 

The control plane, which includes the {ProductShortName} Operators, Istiod, and `ServiceMeshControlPlane`, and the data plane, which includes applications and Envoy proxy, must be in separate namespaces.

File: modules/ossm-control-plane-cli.adoc [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+[id="ossm-control-plane-deploy-cli_{context}"]
+= Deploying the control plane from the CLI
+
+Follow this procedure to deploy a basic `ServiceMeshControlPlane` from the command line.
+
+.Prerequisites
+
+* The {ProductName} Operator must be installed.
+* An account with the `cluster-admin` role.
```
Comment (dgn) [inline modules/ossm-control-plane-cli.adoc:13]: this shouldn't be required once the operator is installed

File: modules/ossm-install-ossm-operator.adoc [hunk 7]
```diff
@@ -4,36 +4,31 @@
 // - service_mesh/v2x/installing-ossm.adoc
 
 [id="ossm-install-ossm-operator_{context}"]
-= Installing the {ProductName} Operator
+= Installing the Operators
 
-.Prerequisites
+To install {ProductName}, install following Operators in this order. Repeat the procedure for each Operator.
```
Comment (dgn) [inline modules/ossm-install-ossm-operator.adoc:9]: ordering shouldn't matter as long as all of them have been installed

File: modules/ossm-member-roll-create.adoc [hunk 8]
```diff
@@ -102,55 +88,19 @@ spec:
...
-  name: mesh-users
-roleRef:
-  apiGroup: rbac.authorization.k8s.io
-  kind: Role
-  name: mesh-user
-subjects:
-- apiGroup: rbac.authorization.k8s.io
-  kind: User
-  name: alice
+$ oc get smmr
```
Comment (dgn) [inline modules/ossm-member-roll-create.adoc:103]: missing `-n istio-system default`

File: modules/ossm-member-roll-modify.adoc [hunk 9]
```diff
@@ -4,7 +4,7 @@
 // * service_mesh/v2x/installing-ossm.adoc
 
 [id="ossm-member-roll-modify_{context}"]
-= Adding or removing projects from the service mesh
+= Modifying projects from the service mesh
```
Comment (dgn) [inline modules/ossm-member-roll-modify.adoc:7]: that title sounds a bit broken now

File: modules/ossm-members.adoc [hunk 10]
```diff
@@ -0,0 +1,54 @@
+// Module included in the following assemblies:
+//
+// * service_mesh/v1x/installing-ossm.adoc
+// * service_mesh/v2x/installing-ossm.adoc
+
+[id="ossm-members_{context}"]
+= Creating the {ProductName} members
+
+`ServiceMeshMember` resources can be created by service mesh users who do not have privileges to add members to the `ServiceMeshMemberRoll` directly. While project administrators are automatically given permission to create the `ServiceMeshMember` resource in their project, they cannot point it to any `ServiceMeshControlPlane` until the service mesh administrator explicitly grants access to the service mesh. Administrators can grant users permissions to access the mesh by granting them the `mesh-user` user role. In this example, `istio-system` is the control plane namespace.
```
Comment (dgn) [inline modules/ossm-members.adoc:9]: I'd start with the value proposition here: 

> `ServiceMeshMember` resources provide a way for service mesh administrators to delegate permissions to add projects to a service mesh, even when the respective users don't have direct access to the service mesh project or member roll`

File: modules/ossm-control-plane-cli.adoc (hunk 6 above)
Comment (neal-timpe) [reply modules/ossm-control-plane-cli.adoc]: @dgn I removed the requirement for an account with the cluster admin role.

The first step to log in "as a user with the `cluster-admin` role". Can I get rid of the as a user with the cluster admin role there too?

File: modules/ossm-members.adoc (hunk 10 above)
Comment (neal-timpe) [reply modules/ossm-members.adoc]: @dgn Thanks! I added that.

Is `apiVersion: rbac.authorization.k8s.io/v1` the correct apiVersion for the RoleBinding?

File: modules/ossm-member-roll-modify.adoc (hunk 9 above)
Comment (neal-timpe) [reply modules/ossm-member-roll-modify.adoc]: I made some changes here.

File: modules/ossm-control-plane-cli.adoc (hunk 3 above)
Comment (neal-timpe) [reply modules/ossm-control-plane-cli.adoc:21]: I hear what you're saying @brian-avery . I suppose it's pretty obvious that you need to log in though. And if you aren't logged in, the terminal would tell you.

**Review by brian-avery — APPROVED** [review]

File: modules/ossm-control-plane-cli.adoc (hunk 3 above)
Comment (neal-timpe) [reply modules/ossm-control-plane-cli.adoc:21]: We're going to leave it as is. It's for orienting our users. It's part of our ccs guidelines.

File: modules/ossm-automatic-sidecar-injection.adoc [hunk 11]
```diff
@@ -6,20 +6,24 @@
...
+----
 
-. Add `sidecar.istio.io/inject` to the configuration YAML with a value of `"true"` as illustrated here:
+. Add `sidecar.istio.io/inject` to the configuration YAML with a value of `"true"` in the `spec.template.metadata.annotations.sidecar.istio/inject` field. See the following example for an app called `sleep`.
 +
-.Sleep test application example
+.Sleep test application example sleep.yaml
 [source,yaml]
 ----
 apiVersion: apps/v1
```
Comment (yxun) [inline modules/ossm-automatic-sidecar-injection.adoc:29]: The sleep yaml sample image: tutum/curl has been deleted from upstream dockerhub. You may replace that with
```
image: curlimages/curl
```
and replace the command with
```
command: ["/bin/sleep","3650d"]
```

File: modules/ossm-install-ossm-operator.adoc [hunk 12]
```diff
@@ -4,36 +4,31 @@
 // - service_mesh/v2x/installing-ossm.adoc
```
Comment (yxun) [inline modules/ossm-install-ossm-operator.adoc:4]: I think OpenShift Elasticsearch operator is an optional operator. By default, we don't need to install Elasticsearch operator before installing OSSM operators. So it's better you update that step "1. OpenShift Elasticsearch" with "1. (Optional) OpenShift Elasticsearch"

File: modules/ossm-members.adoc [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+[source,yaml]
+----
+apiVersion: maistra.io/v1
+kind: ServiceMeshMember
+metadata:
+  name: default
+spec:
+  controlPlaneRef:
+    namespace: istio-system
+    name: minimal-install
```
Comment (yxun) [inline modules/ossm-members.adoc:26]: line 26 name should be consistent across multiple doc pages. In the doc "Deploying the control plane from the CLI" , the SMCP name: basic. So it's better to update line 26 with "name: basic"

**Review by yxun — CHANGES_REQUESTED** [review]
I added three comments and requested changes. Thanks for your help @neal-timpe

**Comment by neal-timpe** [comment]
Thanks @yxun ! I made changes based on your comments. Can you take another look? Do you think the reorganization makes sense?

**Review by yxun — APPROVED** [review]

**Review by gbaufake — APPROVED** [review]
Reviewed @yxun comments and looks good to me.

File: modules/ossm-control-plane-cli.adoc (hunk 6 above)
Comment (dgn) [reply modules/ossm-control-plane-cli.adoc]: Yep, that can be changed to just `log in to the console` -- any account that can create Projects will be sufficient

**Review by dgn — APPROVED** [review]

**Review by pbajjuri20 — APPROVED** [review]
LGTM!
